src/core/config/merge.js

```diff
@@ -15,25 +15,38 @@ import deepExtend from "deep-extend"
 
 const merge = (target, ...sources) => {
   let domNode = Symbol.for("domNode")
-  const sourcesWithoutDomNode = []
+  let primaryName = Symbol.for("primaryName")
+  const sourcesWithoutExceptions = []
 
   for (const source of sources) {
+    const sourceWithoutExceptions = { ...source }
+
     if (Object.hasOwn(source, "domNode")) {
       domNode = source.domNode
-      const sourceWithoutDomNode = { ...source }
-      delete sourceWithoutDomNode.domNode
-      sourcesWithoutDomNode.push(sourceWithoutDomNode)
-    } else {
-      sourcesWithoutDomNode.push(source)
+      delete sourceWithoutExceptions.domNode
+    }
+
+    if (Object.hasOwn(source, "urls.primaryName")) {
+      primaryName = source["urls.primaryName"]
+      delete sourceWithoutExceptions["urls.primaryName"]
+    } else if (source.urls && Object.hasOwn(source.urls, "primaryName")) {
```

Comment:
Yep, your right. I've suggested changes that should make merge survive if `source` is null or undefined and compesation for `source.urls`, which we already use in https://github.com/swagger-api/swagger-ui/pull/9824/files#diff-c0a2622fbb31f1acd04a7359b8482b30f0f0eb432c92baf4eb4c2eff187f677bR46